stl/inc/ranges

```diff
@@ -6821,6 +6822,496 @@ namespace ranges {
+            noexcept(noexcept(_Iterator<true>{_Tuple_transform(_RANGES begin, _Views)})) // strengthened
+            requires(range<const _ViewTypes>&&...) {
+            return _Iterator<true>{_Tuple_transform(_RANGES begin, _Views)};
+        }
+
+        _NODISCARD constexpr auto end() noexcept(_Is_end_noexcept<false>) // strengthened
+            requires(!(_Simple_view<_ViewTypes> && ...)) {
+            if constexpr (!_Zip_is_common<_ViewTypes...>) {
+                return _Sentinel<false>{_Tuple_transform(_RANGES end, _Views)};
+            } else if constexpr ((random_access_range<_ViewTypes> && ...)) {
+                return begin() + static_cast<iter_difference_t<_Iterator<false>>>(size());
+            } else {
+                return _Iterator<false>{_Tuple_transform(_RANGES end, _Views)};
+            }
+        }
+
+        _NODISCARD constexpr auto end() const noexcept(_Is_end_noexcept<true>) // strengthened
+            requires(range<const _ViewTypes>&&...) {
+            if constexpr (!_Zip_is_common<const _ViewTypes...>) {
+                return _Sentinel<true>{_Tuple_transform(_RANGES end, _Views)};
+            } else if constexpr ((random_access_range<const _ViewTypes> && ...)) {
+                return begin() + static_cast<iter_difference_t<_Iterator<true>>>(size());
+            } else {
+                return _Iterator<true>{_Tuple_transform(_RANGES end, _Views)};
+            }
+        }
+
+        _NODISCARD constexpr auto size() noexcept(
+            noexcept(_STD apply(_Size_closure(), _Tuple_transform(_RANGES size, _Views)))) // strengthened
+            requires(sized_range<_ViewTypes>&&...) {
+            return _STD apply(_Size_closure(), _Tuple_transform(_RANGES size, _Views));
+        }
+
+        _NODISCARD constexpr auto size() const
+            noexcept(noexcept(_STD apply(_Size_closure(), _Tuple_transform(_RANGES size, _Views)))) // strengthened
+            requires(sized_range<const _ViewTypes>&&...) {
+            return _STD apply(_Size_closure(), _Tuple_transform(_RANGES size, _Views));
+        }
+
+    private:
+        _NODISCARD static constexpr auto _Size_closure() noexcept {
+            return [](auto... _Sizes) {
+                using _Common_unsigned_type = _Make_unsigned_like_t<common_type_t<decltype(_Sizes)...>>;
+                return (_RANGES min)({static_cast<_Common_unsigned_type>(_Sizes)...});
+            };
+        }
+    };
+
+    template <class... _RangeTypes>
+    zip_view(_RangeTypes&&...) -> zip_view<views::all_t<_RangeTypes>...>;
+
+    template <class... _ViewTypes>
+    inline constexpr bool enable_borrowed_range<zip_view<_ViewTypes...>> = (enable_borrowed_range<_ViewTypes> && ...);
+
+    template <class _ResultType, class... _LHSTupleTypes, class... _RHSTupleTypes>
+        requires(sizeof...(_LHSTupleTypes) == sizeof...(_RHSTupleTypes))
+    _NODISCARD constexpr _ResultType _Zip_get_smallest_iterator_sentinel_distance(

```

Comment:
Is there a reason the declaration and definition are split up for this function?

Reply:
It doesn't need to be, actually. Come to think of it, `_Zip_evaluate_iterator_sentinel_equality` can be defined earlier, too, so I'll move its definition to its first declaration, as well.